extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -0,0 +1,163 @@
+
+    /** @private @const {!Element} */
+    this.documentElement_ = this.getWin().document.documentElement;
+
+    /** @private @const {string} */
+    this.direction_ = this.element.getAttribute('direction');
+
+    /** @private @const {boolean} */
+    this.hasMask_ = false;
+
+    if (this.direction_ != 'left' && this.direction_ != 'right') {
+      this.element.setAttribute('direction', 'left');
+    }
+
+    if (!this.isExperimentOn_) {
+      log.warn(TAG, `Experiment ${EXPERIMENT} disabled`);
+      return;
+    }
+    if (!this.checkWhitelist_()) {
+      return;
+    }
+
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      // Create the mask if the sidebar is rendered in open mode.
+      this.createMask_();
+    }
+
+    this.documentElement_.addEventListener('keydown', event => {
+      // Close sidebar on ESC.
+      if (event.keyCode == 27) {
+        this.close_();
+      }
+    });
+    this.registerAction('toggle', this.toggle_.bind(this));
+    this.registerAction('open', () => {
+      this.open_();
+    });
+    this.registerAction('close', () => {
+      this.close_();
+    });
+  }
+
+  /**
+   * Toggles the open/close state of the sidebar.
+   * @private
+   */
+  toggle_() {
+    if (this.documentElement_.classList.contains('amp-sidebar-open')) {
+      this.close_();
+    } else {
+      this.open_();
+    }
+  }
+
+  /**
+   * Reveals the sidebar.
+   * @private
+   */
+  open_() {
+    this.mutateElement(() => {
+      this.createMask_();
+      this.documentElement_.classList.add('amp-sidebar-open');
+      this.documentElement_.classList.remove('amp-sidebar-closed');
+    });
+  }
+
+  /**
+   * Hides the sidebar.
+   * @private
+   */
+  close_() {
+    this.mutateElement(() => {
+      this.documentElement_.classList.remove('amp-sidebar-open');
+      this.documentElement_.classList.add('amp-sidebar-closed');
+    });
+  }
+
+  /**
+   * Checks if the sidebar only has the whitlisted custom amp- elements.
+   * @returns {boolean} True when only whitelited elements are present.
+   * @private
+   */
+  checkWhitelist_() {
+    const elements = this.element.getElementsByTagName('*');
+    let i = elements.length - 1;
+    while (i >= 0) {
+      const tagName = elements[i].tagName;
+      if (tagName.indexOf('AMP-') == 0) {
+        const isWhiteListed = assert(
+            WHITELIST_.indexOf(tagName) >= 0,
+            '%s can only contain the following custom tags: %s',
+            this.element, WHITELIST_);
+        if (!isWhiteListed) {
+          return false;
+        }
+      }
+      i--;
+    }
+    return true;
+  }
+
+  /**
+   * @private
+   */
+  createMask_() {
+    if (this.hasMask_) {
+      return;
+    }
+    const mask = document.createElement('div');
+    mask.setAttribute('class', '-amp-sidebar-mask');
+    mask.addEventListener('click', () => {
+      this.toggle_();
+    });
+    this.element.parentNode.appendChild(mask);
```

Comment:
The masking is happening on the body (parent) and not inside the sidebar itself - i think it is more semantic/appropriate here.